### DISCUSSION TITLE
ayuda por mas documentacion faas, lambda con cotalker, y solo unas apis funcionan

### DISCUSSION THREAD

**Discussion by z0rro2k** [Cotalker/documentation#456] [2. Questions and Answers] [answered]
Hola Cotalker

  Podran compartir un documento paso a paso para usar faas, llamada de red. necesitamos generar un reporte pdf, y enviarlo por email de forma programada.
  Se nos ha sugerido usar faas, llamada red/api.

  Me han dado algunas guias, no tengo experiencia en node, javascript/java, ni funcion lambda
  Si tengo experiencia en html/postman/ y algo en webservices
  No he podido darle en el clavo al faas que deberia recibir parametros (datos de incidentes) para obtener informacion y mostrarlos en la salida

  ***En mi caso, la documentacion es insuficiente, y no he encontrado datos claros o paso a paso, busco/necesito un manual corta palos para poder avanzar de forma autonoma

  ***Otro punto importante, en la documentacion de endpoints, al ingresar el token y probar las distintas apis (ahora GET), solo algunas fucnionan y la mayoria no. las pruebo en la misma pagina de Cotalker API y postman, pero solo unos pocos traen datos, la mayoria da error

  Gracias por su ayuda

**Comment by guillevalin** [comment, marked as answer]
Hola,

Respecto al FaaS, la verdad es que para resolverlo debes utilizar programación pura, y no es algo en particular de Cotalker, de alguna forma tienes que tener un endpoint que genere una página HTML con valores dinámicos que provienen de "query parameters" o incluso de un "body" (si el endpoint es POST).

Yo recomendaría que tomaras el proyecto base que ya genera una página HTML en el la ruta GET / y hagas pruebas mandando los valores por query parameters, ?titulo=[algo]&mensaje=[otra cosa] etc. Luego de probar que funciona bien y subirlo a AWS Lambda, podrás usar ese endpoint en el bot "Crear PDF" concatenando los COTLang de los valores que quieres mostrar en los parámetros que configuraste.

Respecto a los endpoints que no te funcionan, podrían ser por parámetros o permisos del token así que te pediría más información para poder ayudarte mejor en esos otros puntos.
